tinygrad/runtime/lib.py

```diff
@@ -37,7 +37,7 @@ class RawBufferMapped(RawBufferCopyIn):
   def _buffer(self) -> memoryview: raise NotImplementedError("must be implemented")
   # NOTE: this metadata prevents the backing buffer from being freed. hack can be removed with PEP688
   def buffer_view(self) -> np.ndarray: return np.frombuffer(self._buffer(), dtype=np.dtype(self.dtype.np, metadata={"backing": self}), count=self.size)  # type: ignore
-  def toCPU(self) -> np.ndarray: return self.buffer_view().copy() # Need a copy, since jit will write to the same buffer.
+  def toCPU(self) -> np.ndarray: return self.buffer_view().copy() if hasattr(self, '_needs_explicit_copy') else self.buffer_view() # JIT sets this to not overwrite buffers holded by CPU.
```

Comment:
The should just shouldn't be here. Can the JIT make the copy?

Reply:
Yeah, will think on this